wgpu-native/src/lib.rs

```diff
@@ -166,46 +169,117 @@ pub struct ByteArray {
     pub length: usize,
 }
 
-pub type InstanceId = hub::Id;
+macro_rules! define_id {
+    ($i:ident) => {
+        transparent!($i);
+        typed_id!($i);
+        new_id!($i);
+    }
+}
+
+
+macro_rules! transparent {
+    ($i:ident) => (
+        #[repr(transparent)]
+        #[derive(Clone, Copy, Debug, Hash, PartialEq)]
+        #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+        pub struct $i(Id);
+    )
+}
+
+pub trait TypedId:From<Id> {
+    fn raw(&self) -> Id;
+}
+
+macro_rules! typed_id {
+    ($i:ident) => ( 
+        impl From<Id> for $i {
+            fn from(id:Id) -> $i {
```

Comment:
let's have this method just as `fram_raw` instead of requiring `From<Id>`

Reply:
Now that `IdentityManager` is strongly typed, a `from_raw()` method is not necessary so I delete it